Example-CustomUserGroups-REXX/README.md

```diff
@@ -0,0 +1,500 @@
+
+## Invocation
+**Syntax**:
+```
+USRGRP01 [-C CPCName] [-L LPARName] [-G CustomUserGroup] [-A Action {ADD, REMOVE, LIST}] [-I] [-V] [-H]
+```
+
+ Optional Input Parameters:<br>
+  - CPCName  - The name of the CPC to query, the default is the local CPC if not specified or provided<br>
+  - LPARName - The name of the LPAR to query and target, the default is
+                 the local LPAR if not specified or provided<br>
+  - UserGroupName - The name of the Custom User Group to target<br>
+  - Action        - The action to take against the Custom User Group
+                    limited to ADD, REMOVE, or LIST<br>
+  - -I            - indicate running in ISV REXX environment, default
+                    is TSO/E<br>
+  - -V            - turn on additional verbose JSON tracing<br>
+  - -H            - Display sample parameters and how to invoke the sample<br>
+
+**Note**: If no arguments are supplied then the sample will default to listing all of the Custom User Groups on the local CPC.
+
+**Sample Invocation in TSO:**
+
+USRGRP01 has been copied into data set HWI.USER.REXX <br>
+
+  List all Custom User Groups on the specified CPC:<br>
+  - Local CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)'
+    ```
+
+  - Remote CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-C CPC1 -L LPAR1'
+    ```
+
+  List the members of the TEST Custom User Group:
+  - Local CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-G TEST'
+    ```
+
+  - Remote CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-C CPC1 -L LPAR1 -G TEST'
+    ```
+
+  Add an LPAR as a new member to the TEST Custom User Group
+
+  - Local CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-G TEST -A ADD'
+    ```
+
+  - Remote CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-C CPC1 -L LPAR1 -G TEST -A ADD'
+    ```
+
+  Remove an LPAR member from the TEST Custom User Group
+
+  - Local CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-G TEST -A REMOVE'
+    ```
+
+  - Remote CPC and LPAR:
+    ```
+    ex 'HWI.USER.REXX(USRGRP01)' '-C CPC1 -L LPAR1 -G TEST -A REMOVE'
+    ```
+
+**Sample Batch Invocation via JCL:**
+USRGRP01 has been copied into data set HWI.USER.REXX
+
+```
+ //USRGRP01 JOB ,
+ // CLASS=J,NOTIFY=&SYSUID,MSGLEVEL=1,
+ //  MSGCLASS=H,REGION=0M,TIME=1440
+ //STEP1    EXEC PGM=IKJEFT01,DYNAMNBR=20
+ //SYSUDUMP DD SYSOUT=(H,,STD)
+ //SYSTSPRT DD SYSOUT=(H,,STD)
+ //SYSTSIN  DD * UB
+ PROFILE NOPREFIX
+ EX 'HWI.USER.REXX(USRGRP01)'  -
+ 'LOCAL_CPC LOCAL_LPAR TESTGRP ADD'
+ /*
+ ```
+ - exec is running in a TSO/E rexx environment
+
+## Example Generated Output
+
+### Success Path for List Custom User Groups Invocation
+```
+------->
+GET request being made....
+URI: /api/cpcs?name=CPC1
+ENCODING: 0
+TIMEOUT: 0
+
+Rexx RC: (0)
+HTTP Status: (200)
+SE DateTime: (Tue, 19 Jul 2022 19:24:18 GMT)
+SE requestId: (Sxabd34565-df4e-33fg-1010-99574c560a56.12 Rx3)

```

Comment:
general comment
- replace all SE requestID content with something fake
- replace all uuid's in the uri's with something fake